client/app/visualizations/counter/utils.test.js

```diff
@@ -0,0 +1,168 @@
+import { getCounterData } from './utils';
```

Comment:
Not able to stub `getCounterData` and other private functions in utils at the moment. Will be addressed in another PR.

Reply:
Why would you want to stub it?

Reply:
I meant to say '`getRowNumber` and other private functions'. Stubbing (mocking) would depend on the testing strategy though.

Atm all these 'private' functions including `getRowNumber`, `numberFormat`, `formatValue` etc: 
1. Contain logic and are not covered. This is fixable by simply exporting and running cases on them.
2. From a puristic perspective, when executing tests on `getCounterData` one would want to mock the calls to these private functions to isolate the test subject. Otherwise it's more of an integration testing (which is not too bad anyway). This type of mocking is not an easy one in React as it does not have dependency injection and available solutions [are](https://stackoverflow.com/questions/45111198/how-to-mock-functions-in-the-same-module-using-jest) [cumbersome](https://medium.com/@DavideRama/mock-spy-exported-functions-within-a-single-module-in-jest-cdf2b61af642) or would require coding style.

I assume it's ok to test on integration level and gradually add tests for other functions.

Reply:
> I meant to say '`getRowNumber` and other private functions'.

Oh, ok. This makes sense now :-)

> I assume it's ok to test on integration level and gradually add tests for other functions.

It's indeed ok. We're not purists here but rather pragmatists :-) So even having just a test for the `getCounterData` function is enough considering its scope/size.

Reply:
> We're not purists here but rather pragmatists :-)

Nice one. Aligned.